kratos/includes/properties.h

```diff
@@ -278,31 +293,383 @@ class Properties : public IndexedObject
         return result_key;
     }
 
+    /**
+     * @brief This method returns the number of subproperties
+     * @return The current number of subproperties
+     */
+    std::size_t NumberOfSubproperties()
+    {
+        return mSubPropertiesList.size();
+    }
+
+    /**
+     * @brief This method insert a new property into the list of subproperties
+     * @param pNewSubProperty The new property to be added
+     */
+    void AddSubProperty(Properties::Pointer pNewSubProperty)
+    {
+        KRATOS_ERROR_IF(HasSubPropertiesById(pNewSubProperty->Id())) << "SubProperties with Id: " << pNewSubProperty->Id() << " already exists. In each level the Id of properties (new or existing) should be unique" << std::endl;
+        mSubPropertiesList.insert(mSubPropertiesList.begin(), pNewSubProperty);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The pointer to the subproperty of interest
+     */
+    KRATOS_DEPRECATED_MESSAGE("This is legacy version. Use pGetSubPropertiesById") Properties::Pointer pGetSubProperty(const IndexType SubPropertyIndex)
+    {
+        return pGetSubPropertiesById(SubPropertyIndex);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id (constant version)
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The pointer to the subproperty of interest
+     */
+    KRATOS_DEPRECATED_MESSAGE("This is legacy version. Use pGetSubPropertiesById") Properties::Pointer pGetSubProperty(const IndexType SubPropertyIndex) const
+    {
+        return pGetSubPropertiesById(SubPropertyIndex);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    KRATOS_DEPRECATED_MESSAGE("This is legacy version. Use GetSubPropertiesById") Properties& GetSubProperty(const IndexType SubPropertyIndex)
```

Comment:
I really would not add sth new to the core that is already deprecated
I know it is annoying that those decisions are taking long, but this is a search-replace to update it

Reply:
@AlejandroCornejo among others already did many developments using this branch as basis

Reply:
I know, but changing it will be easy, only the name of the fct is different